File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.java
import java.util.Arrays;
class Solution {
    public int solution(int[][] sizes) {
        int answer = 0, temp, WidthMax = 0, LengthMax = 0;
        
        for(int i = 0; i<sizes.length; i++){
            if(sizes[i][0] < sizes[i][1]){
                temp = sizes[i][0];
                sizes[i][0] = sizes[i][1];
                sizes[i][1] = temp;
            }
            
            if(sizes[i][0] > WidthMax) WidthMax = sizes[i][0];
            if(sizes[i][1] > LengthMax) LengthMax = sizes[i][1];
        }
        
        
        
        return answer = WidthMax*LengthMax;
    }
}

File: 프로그래머스/1/86491. 최소직사각형/최소직사각형.py
def solution(sizes):
    answer = 0
    max_width, max_height = 0, 0
    
    for (width, height) in sizes:
        max_width = max(max(width, height), max_width)
        max_height = max(min(width, height), max_height)
        
    return max_width * max_height